src/ngNotificationsBar.js

```diff
@@ -153,6 +158,10 @@
 					notificationHandler(event, data, 'success');
 				});
 
+				scope.$on('notifications:closeAll', function () {
+					notifications.length = 0;
```

Comment:
Yep that's the reason that I went with that strategy over the one you initially suggested - I think we actually ended up at the same SO thread.